kubeyard/commands/shell.py

```diff
@@ -47,7 +51,19 @@ def run_default(self):
     @cached_property
     def pod_name(self) -> str:
         if self.pod:
-            return self.pod
+            all_pods = sh.kubectl.get.pods('-o', 'jsonpath={.items[*].metadata.name}').split()
+            # Exact match
+            if self.pod in all_pods:
+                return self.pod
+            # Starting-with match
+            pods = [pod for pod in all_pods if pod.startswith(self.pod)]
+            if len(pods) == 1:
+                return pods[0]
+            elif len(pods) == 0:
+                raise CommandException(f"Not found pod equal or starting with '{self.pod}'")
+            else:
+                logger.warning("Found more than one pod starting with '%s'" % self.pod)
+                return pods[0]
```

Comment:
`if len(pods) == 0:
    raise CommandException(f"Not found pod equal or starting with '{self.pod}'")
elif len(pods) > 1:
    logger.warning(f"Found more than one pod. Using '{pods[0]}'")
return pods[0]`
Implementation seems to be more readable

Reply:
`len(pods) > 1`  is implicated by other condition, so how about putting comment instead of repeating it?

Reply:
Where is this other condition?

Reply:
I mean two `if` just above.

Reply:
Solved as you proposed